### ISSUE TITLE
[mac1] Dropbox hasn't completed synch for a few months

### ISSUE THREAD

**Issue by 130s** [kinu-garage/hut_10sqft#467]
After I tinkered with some mass downloads on Microsoft Sharepoint client, Dropbox started getting stuck with a few dozens of thousands of files. IIRC, it was sometime in June or even May, so it's been 5 months or so since mac1 hasn't finished synching.

**Comment by 130s** [comment]
Found on Dropbox' desktop client that there is "View issues" button. Clicking there showed a list of files that are having an issue in synching, which lead me to https://help.dropbox.com/installs-integrations/sync-uploads/stuck-syncing#permission that suggests the following. So I ran them.
```
#sudo chflags -R nouchg ~/data/Dropbox ~/.dropbox ~/.dropbox-master
sudo chflags -R nouchg ~/data/Dropbox ~/.dropbox
sudo chown "$USER" "$HOME"
sudo chown -R "$USER" ~/data/Dropbox ~/.dropbox
sudo chmod -RN ~/.dropbox ~/data/Dropbox
chmod -R u+rw ~/data/Dropbox ~/.dropbox
```
We'll see how things go.

**Comment by 130s** [comment]
No good.

Followed advanced re-installation https://help.dropbox.com/installs-integrations/desktop/advanced-reinstall. See if this changes.

**Comment by 130s** [comment]
Not sure if it's working or not. I just want to see Dropbox complete sync even the subset of sync list. So eliminated the Onedrive folder that is suspicious root cause for now.

**Comment by 130s** [comment]
I'm seeing the files that are downloaded from Microsoft Onedrive client cannot be deleted on `Finder`, even after it asked admin password. On commandline, those same files can be removed after `sudo`.

I do see ["extended attributes" (unix.stackexchange.com)](https://unix.stackexchange.com/questions/10/what-does-the-mean-in-ls-l), although I do not yet get what each of those attrs mean and how they affect Dropbox (why Finder can't delete after passing admin password?).

**Comment by 130s** [comment]
Example of Onedrive file:
```
$ ls -l@ 362_default-config.tgz 
-rwxrwxrwx@ 1 n130s  staff  27378 Sep 24 19:24 362_default-config.tgz
	com.apple.LaunchServices.OpenWith	  118 
	com.apple.fileutil.PlaceholderData	  336 
	com.apple.metadata:com_apple_backup_excludeItem	   61 
	com.microsoft.OneDrive.RecallOnOpen
```
Btw I do find extended attrs with other files. E.g. the one synched with Google Photo desktop client:
```
$ ls -l@ adies_skiing.jpg 
-rw-r--r--@ 1 n130s  staff  439598 Dec  8  2017 adies_skiing.jpg
	com.dropbox.attributes	    83 
	com.dropbox.attrs	    26 
```
Again I don't know what these mean and how I can get a control over them.

----

Then, looking at older folders where I didn't see sync issue, I do not see the `OneDrive` specific extended attribute.
```
$ ls -l@ 2019-12-26-17-50-12.bag 
-rwxrwxrwx@ 1 n130s  staff  14314 Dec 26  2019 2019-12-26-17-50-12.bag
	com.apple.fileutil.PlaceholderData	  336 
	com.dropbox.attrs
```
Btw on Ubuntu I don't even see the extended attr (is the concept available on OSX only?). For the same file,
```
$ ll 2019-12-26-17-50-12.bag 
-rwxr--r-- 1 n130s n130s 14K Dec 26  2019 2019-12-26-17-50-12.bag*
```

**Comment by 130s** [comment]
Getting tired (finally after a few months...), I deleted the previous Onedrive folder locally and restarted cloning. OneDrive app remains to s@ck after a few months, constantly stopping to respond, and failed to handle selective folder sync. We'll see.

**Comment by 130s** [comment]
mac1 seems still struggling.

Btw, almost everyday after I left mac1 on for hours, I found dropbox is gone...So many troubles.

**Comment by 130s** [comment]
> mac1 seems still struggling.

Yes, still is.

> Btw, almost everyday after I left mac1 on for hours, I found dropbox is gone...So many troubles.

This is not the case for the past few weeks.

Anyways, I just did the following again. We'll see.

```
$ sudo chflags -R nouchg ~/data/Dropbox ~/.dropbox ~/.dropbox-master
chflags: /Users/n130s/.dropbox-master: No such file or directory

sudo chown "$USER" "$HOME"
sudo chown -R "$USER" ~/data/Dropbox ~/.dropbox

sudo chmod -RN ~/.dropbox ~/data/Dropbox
chmod -R u+rw ~/data/Dropbox ~/.dropbox

echo "manually rebooting OS"
```

I don't know if this is related or not but I see before reboot completes.
```
 [Restored Nov 11, 2020 7:36:15]
Last login: Wed Nov 11 07:36:13 on ttys004
Restored session: Wed Nov 11 07:34:38 EST 2020

The default interactive shell is now zsh.
To update your account to use zsh, please run `chsh -s /bin/zsh`.
For more details, please visit https://support.apple.com/kb/HT208050.
```

**Comment by 130s** [comment]
`zsh` warning doesn't seem to be malignant esp. for Dropbox context (thought I'm not 100% sure) https://apple.stackexchange.com/questions/371997/suppressing-zsh-verbose-message-in-macos-catalina so I leave it for now.

**Comment by 130s** [comment]
STILL no progress.

Versions
- dropbox v112.4.321
- macOS Catalina 10.15.7

Synopsis
- Dropbox on this computer keeps "synching". This has been 6+ months.
   - All other computers (Ubuntu Linux) seem to sync well.
- From desktop app, clicking the "sync issue" then I see several hundreds of errors. AFAICT, all link to [Dropbox is stuck syncing, won't open, or shows an error](https://help.dropbox.com/installs-integrations/sync-uploads/stuck-syncing#permission).
   - Strange that these files are under the folder that I once had Dropbox sync, but not any more (I excluded the folder by Selective Sync). On Finder, on the icon of these files I still see DB's sync status sign (some look synching, some look crossed out).
- I've also seen several times that while I'm looking at the list of sync errors, a pop up window says Dropbox is restarting, and the windows that showed sync error disappers.

Trying once again the suggestion on [the link](https://help.dropbox.com/installs-integrations/sync-uploads/stuck-syncing#permission).

```
linguinne-mac1:~ nlinguinne$ sudo chflags -R nouchg ~/baa/Dropbox ~/.dropbox ~/.dropbox-master
Password:
chflags: /Users/nlinguinne/.dropbox-master: No such file or directory

linguinne-mac1:~ nlinguinne$ ls ~/.dropbox/
Crashpad/        dropbox.pid      host.db          instance1/       logs/            metrics/
QuitReports/     events/          info.json        instance_db/     machine_storage/ unlink.db

echo "Skipping this as I already see all files and folders are owned by $USER
# sudo chown "$USER" "$HOME"
linguinne-mac1:~ nlinguinne$ ls -l $HOME
total 9320
(cut)

sudo chown -R "$USER" ~/baa/Dropbox ~/.dropbox

sudo chmod -RN ~/.dropbox ~/baa/Dropbox
chmod -R u+rw ~/baa/Dropbox ~/.dropbox
```
We'll see. If this won't change anything I'll ask on Dropbox forum.

**Comment by 130s** [comment]
Finally I requested a support in the paid user channel:

> I've been trying to resolve by myself using public community but I've reached the point where I need to fix. So as a paid user I'm asking for the help.
> 
> # Problem:
> - Dropbox app keeps showing "sync" sign. My note says it's been like this since May 2020.
>    - If helpful, I've been publicly traking my progress at https://github.com/130s/hut_10sqft/issues/467
>    - I've tried a number of times "View sync issues" -> "fix issues". No good.
> 
> # Other things noticed:
> - Also, though I'm not sure if this is related, but while I have "View sync issues" window opened, Dropbox app tends to restart.
>    - This doesn't concern me as long as the sync issue can get resolved.
> - On other devices (Ubuntu, Android) I don't seem to have sync issues.
> 
> # Environment:
> - Mac OS Catalina (during 7 months I might have upgraded from a previous version).
> - Dropbox app version: 115.4.601
> - No firewall or any special security measure is taken to my knowledge other than what Mac OS does.

**Comment by 130s** [comment]
Support can see I have more files than 300K files, which is known to slow down Dropbox, and suggested Selective Sync to reduce the number of files on this host.

My response:

> Regarding Selective Sync, I forgot to mention that:
> - On this problematic host, I've unchecked some folders that contain large number of files, as an attempt to resolve the issue.
> - Still, "View sync issues" from Dropbox app shows the files that are stored in the folders that are unchecked in Selective Sync.
> 
> With that, I don't know if messing further with Selective Sync would make much difference. That said yes I can try something extreme (e.g. unchecking all).

I went ahead:
- moving the folders in question in my response above to somewhere outside of Dropbox' sync path.
- Reduced on Selective Sync the folders to sync to very little.
   - Since some folders overlap with Google's Backup and Sync, which is functional so the files under any unchecked folders on Dropbox would be deleted on G-Drive, which is known to cause a whole new issue https://github.com/130s/hut_10sqft/issues/501, I temporarilly closed Backup and Sync.

**Comment by 130s** [comment]
Synched **FINALLY**. See the email I sent to tech support.

On Mon, Feb 8, 2021 at 12:21 PM Isao Yanagisawa (Isaac Saito) <130s@2000.jukuin.keio.ac.jp> wrote:
>
> Dropbox app finished sync finally!
>
> I can't tell what eventually worked. But I assume the removal of the folder at path A is an unblocker.
>
> Now I checked back many folders on Selective Sync so that the I'm syncing many more files than I was when you suggested to reduce the number of sync, but I see Dropbox is still syncing.
>
> Please close the case. If this is a known issue, I'd like to see where it is logged. But if it's not known, I'd suggest to file an issue.
>
> - ---  Isaac Saito (Isao Yanagisawa)
>
>
> On Mon, Feb 8, 2021 at 7:32 AM Isao Yanagisawa (Isaac Saito) <130s@2000.jukuin.keio.ac.jp> wrote:
>>
>> It's been almost 45 hours since I made the last change. In my understanding, the number of files Dropbox desktop app tracks is reduced way below 300K items. I also placed some files as a test that only exist on this Mac in question (so if I see that on dropbox.com, at least these test files are synched from Mac to cloud). What I see:
>>
>> - Icon of Dropbox desktop app at the desktop bar stll shows "synching".
>> - On Dropbox desktop app,
>>
>>    - "Syncing 188,645 files   99.7GB (1 sec)" status bar at the bottom (I kept paying attention to these numbers over hours and they hadn't changed). Clicking that bar, it shows "Checking for updates. Please hold while we make sure everything is up to date".
>>    - "Dropbox Sync issues" shows "No errors", which is different from what I used to see before I started this support request.
>>
>> - On Finder (Exlorer equivalent in Windows?) I see the "synching" icons on many folders.
>> - Test files are not yet present on dropbox.com.
>>
>> With taking almost 2 days, I'd call it "not working".
>>
>> Then I found this.
>> There's a folder (that contained files as many as 60K with 7GB total) that got moved to path A to path B, both of which are under dropbox sync path. On this Mac, I have neither path A nor B synched in Selective Sync. But on Finder I still saw it at path A, and Finder showed the "syncing" icon. I then removed this folder, then the "Syncing X files   Y GB" finally started changing (they first went up, then now decreasing).
>>
>> I'll see how this goes.

I have yet to sync Google Drive (as I temporarilly closed it https://github.com/130s/hut_10sqft/issues/467#issuecomment-774283699). To do that I'll sync as many folders as possible on Mac in the Dropbox path, so that Backup and Sync doesn't delete undesired files. Also as a backup measure for undesired deletion, I emptied Trash on G-Drive (so any files that are found in Trash frmo now on are the ones that are undesired deleted so that we can restore, unless I explicitly delete).

**Comment by 130s** [comment]
Dropbox on Mac generated dozens of "conflicted copy" folders. Usually this is bad I feel, but I pardoned it for doing that just for today as it's far better than not syncing at all. I manually deleted all those duplicated folders on 1) `find . -iname "*conflicted copy*"` on Linux then 2) manually deleted on dropbox.com. Duplicated files are deleted by my custom bash command.

Waiting for G-Drive desktop client, which I kept it off for a few days so it has files to sync that seems to be taking a while, to finish syncing.
